components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -709,97 +728,101 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
+                                                    Integer limit, String organizationId, String sortOrder,
+                                                    List<ExpressionNode> expressionNodes,
+                                                    List<ExpressionNode> parentIdExpressionNodes,
+                                                    String applicationAudience)
             throws OrganizationManagementServerException {
 
-        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
-        appendFilterQuery(expressionNodes, filterQueryBuilder);
-        Map<String, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
-        List<String> timestampTypeAttributes = filterQueryBuilder.getTimestampFilterAttributes();
-
-        FilterQueryBuilder parentIdFilterQueryBuilder = new FilterQueryBuilder();
-        appendFilterQueryForParentId(parentIdFilterQueryBuilder, parentIdExpressionNodes);
-        Map<String, String> parentIdFilterAttributeValueMap = parentIdFilterQueryBuilder.getFilterAttributeValue();
-        String parentIdFilterQuery = parentIdFilterQueryBuilder.getFilterQuery();
+        FilterQueryBuilder filterQueryBuilder = buildFilterQuery(expressionNodes);
+        FilterQueryBuilder parentIdFilterQueryBuilder = buildParentIdFilterQuery(parentIdExpressionNodes);
 
-        String sqlStmt;
-        String getOrgSqlStmtTail;
-        if (!CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
-            getOrgSqlStmtTail = authorizedSubOrgsOnly
-                    ? StringUtils.isNotBlank(applicationAudience)
-                    ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL
-                    : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL
-                    : GET_ORGANIZATIONS_TAIL;
-
-            if (isOracleDB()) {
-                getOrgSqlStmtTail = authorizedSubOrgsOnly
-                        ? StringUtils.isNotBlank(applicationAudience)
-                        ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL_ORACLE
-                        : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL_ORACLE
-                        : GET_ORGANIZATIONS_TAIL_ORACLE;
-            } else if (isMSSqlDB()) {
-                getOrgSqlStmtTail = authorizedSubOrgsOnly
-                        ? StringUtils.isNotBlank(applicationAudience)
-                        ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL_MSSQL
-                        : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL_MSSQL
-                        : GET_ORGANIZATIONS_TAIL_MSSQL;
-            }
+        String userID =  getUserId();
+        String sqlStmt = prepareGetOrganizationQuery(authorizedSubOrgsOnly, recursive, sortOrder, applicationAudience,
+                        filterQueryBuilder, parentIdFilterQueryBuilder, userID);
 
+        List<String> permissions;
+        String permissionPlaceholder;
+        if (Boolean.TRUE.equals(CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME)) {
```

Comment:
```suggestion
        if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
```
fix other places as well

Reply:
let's remove the logics related to ENABLE_LEGACY_AUTHZ_RUNTIME true cases

We are in a process of cleaningup the legacy runtime code base. So through this PR itself let's clean the usages in DAO layer

Reply:
All the `CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME` boolean checks are now removed. Only the modern runtime logic remains.